### PR TITLE
Implement default version for template and default upgrade version for template version

### PR DESCRIPTION
**Pull request by amy**
https://github.com/rancher/rancher/issues/8998

### CODE REVIEW

File: model/version.go [hunk 1]
```diff
@@ -32,6 +35,8 @@ type Version struct {
 	Questions []Question `sql:"-"`
 }
 
+type Versions []Version
```
Comment (amy) [inline model/version.go]: Created a type that implements sort.Interface

File: service/utils.go [hunk 2]
```diff
@@ -169,11 +200,24 @@ func versionResource(apiContext *api.ApiContext, catalogName string, template mo
...
+			upgradeVersions = append(upgradeVersions, upgradeVersion)
+
+			if template.DefaultVersion == upgradeVersion.Version {
+				links["defaultUpgradeVersion"] = URLEncoded(link)
+			}
+		}
+	}
+
+	if _, ok := links["defaultUpgradeVersion"]; !ok {
+		defaultUpgradeVersionLink := defaultUpgradeVersionLink(upgradeVersions, catalogName, template, apiContext)
```
Comment (amy) [inline service/utils.go]: Picks the latest valid upgrade version link if the default version is not a valid upgrade.

**Comment by amy** [comment]
cc/ @cjellick @vincent99

File: scripts/test [hunk 3]
```diff
@@ -15,4 +15,5 @@ go test ${RACE} -cover -tags=test ${PACKAGES}
 ./bin/rancher-catalog-service --sqlite --config scripts/test-repo.json -p 8088 &
 trap "kill $!" exit
 cd integration
-tox -e flake8,py27
+tox -e py27 -- -s
```
Comment (joshwget) [inline scripts/test]: What's the purpose for this change?

Comment (StrongMonkey) [inline scripts/test]: (hiding all flake8 errors)
